medcat/utils/checkpoint.py

```diff
@@ -60,65 +54,118 @@ def count(self) -> int:
+        self._file_paths = ckpt_file_paths
+        self._count = count
+
+    def purge(self) -> List[str]:
+        r'''
+        Remove all checkpoint files from the checkpoint directory.
+
+        Returns:
+            A list of paths to the removed files
+        '''
+        ckpt_file_paths = self._get_ckpt_file_paths(self._dir_path)
+        removed = []
+        for path in ckpt_file_paths:
+            if os.path.isfile(path):
+                os.remove(path)
+                removed.append(path)
         self._file_paths = []
         self._count = 0
-
-    def save_metadata(self) -> None:
-        metadata_file_path = os.path.join(os.path.abspath(self._dir_path), "checkpoint-metadata.json")
-        if self._metadata is not None:
-            with open(metadata_file_path, "w") as f:
-                f.write(jsonpickle.encode(self._metadata))
-            self.log.info("Checkpoint metadata saved: %s", metadata_file_path)
-        else:
-            raise Exception("Checkpoints metadata not found.")
+        return removed
 
     def save(self, cdb: CDB, count: int) -> None:
+        r'''
+        Save the CDB as the latest checkpoint.
+
+        Args:
+            cdb (medcat.CDB):
+                The input MedCAT CDB object
+            count (count):
+                The number of the finished steps
+        '''
         ckpt_file_path = os.path.join(os.path.abspath(self._dir_path), "checkpoint-%s-%s" % (self.steps, count))
         while len(self._file_paths) >= self._max_to_keep:
             to_remove = self._file_paths.pop(0)
             os.remove(to_remove)
         cdb.save(ckpt_file_path)
-        self.log.info("Checkpoint saved: %s", ckpt_file_path)
+        self.log.debug("Checkpoint saved: %s", ckpt_file_path)
         self._file_paths.append(ckpt_file_path)
         self._count = count
 
-    async def save_async(self, cdb: CDB, count: int) -> None:
-        ckpt_file_path = os.path.join(os.path.abspath(self._dir_path), "checkpoint-%s-%s" % (self.steps, count))
-        await cdb.save_async(ckpt_file_path)
-        self.log.info("Checkpoint saved: %s", ckpt_file_path)
-        self._file_paths.append(ckpt_file_path)
-        self._file_paths.sort(key=lambda f: self._get_steps_and_count(f)[1])
-        self._count = count
-        while len(self._file_paths) > self._max_to_keep:
-            to_remove = self._file_paths.pop(0)
-            os.remove(to_remove)
-
     def populate(self, cdb: CDB) -> None:
+        r'''
+        Populate the latest checkpoint to a CDB.
+
+        Args:
+            cdb (medcat.CDB):
+                The input MedCAT CDB object
+        '''
         if not self._file_paths:
-            raise Exception("Checkpoints not found. You need to restore or train from scratch.")
+            raise Exception("Cannot populate the model. Make sure the checkpoint is restored beforehand.")
         cdb.load(self._file_paths[-1])
 
+    def restore_and_populate(self, cdb: CDB) -> None:
```

Comment:
comment as above, maybe just call this load, and load_latest with those implementations in here?

Reply:
Good point. Renamed it to `restore_latest_cdb`.